mocks/mocks.py

```diff
@@ -108,26 +110,31 @@ def scale_light_by_distance(particle_positions,
     return particle_positions.value, particle_values
 
 
-def prepare_mocks(config_location='config_example.yaml',
+def prepare_mocks(config,
```

Comment:
This function is doing 4 unrelated things. I would suggest splitting this into at least three functions.

One fully parses the config, i.e. loads the config file with `mocks.utils.load_yaml_config`, then goes through _all_ the parameters (not just `galaxy_coordinates` & `observation_time` as now) and applies any keyword arguments that override the config file values, gets the remaining values from the config file, adds default values for any parameters that are missing, and does any required unit conversions, etc.  This function returns the fully populated config dict for all the other functions to use.  The other functions then don't have to do any fiddling around with overriding values, default values, unit conversions, etc.

The second function takes the config as its only argument, loads the simulation data, and returns it. This would be the place to do 32 bit to 64 bit float type conversion, when required.

The calls to `compute_total_mass` and `compute_apparent_ABmag` also don't belong here.  You could create a 3rd function that callsthem, using values from `config`,  and returns the results, or you could do those calls in the main function.